faros_event.sh

```diff
@@ -326,16 +334,171 @@ function processEventTypes() {
+                status_details=$run_status_details
+            else
+                status_details=""
+            fi
+            if ! [ -z "$has_run_start_time" ] &&
+               ! [ -z "$has_run_end_time" ]; then
+                start_time=$(unix_seconds_to_iso8601 $run_start_time)
+                end_time=$(unix_seconds_to_iso8601 $run_end_time)
+
+                cicd_Build_with_start_end=$( jq -n \
+                                --arg run_status "$run_status" \
+                                --arg run_status_details "$run_status_details" \
+                                --arg run_start_time "$start_time" \
+                                --arg run_end_time "$end_time" \
+                                '{ 
+                                    "run_status": {"category": $run_status, "detail": $run_status_details},
+                                    "run_start_time": $run_start_time,
+                                    "run_end_time": $run_end_time,
+                                }'
+                            )
+                cicd_Build_with_start_end=$(concat "$cicd_Build_with_start_end" "$buildKey")
+                make_mutation cicd_build_with_start_end "$cicd_Build_with_start_end"
+            else
+                cicd_Build=$( jq -n \
+                                --arg run_status "$run_status" \
+                                --arg run_status_details "$run_status_details" \
+                                '{
+                                    "run_status": {"category": $run_status, "detail": $run_status_details},
+                                }'
+                            )
+                cicd_Build=$(concat "$cicd_Build" "$buildKey")
+                make_mutation cicd_build "$cicd_Build"
+            fi
+
+            cicd_Pipeline=$(jq \
+            '{data_run_pipeline,data_run_organization,data_run_source}' <<< $flat
+            )
+            make_mutation cicd_pipeline "$cicd_Pipeline"
+
+            cicd_Organization_from_run=$(jq \
+            '{data_run_organization,data_run_source}' <<< $flat
+            )
+            make_mutation cicd_organization_from_run "$cicd_Organization_from_run"
+        fi
+    else
+        make_mutation cicd_artifact "$artifact_key"
+    fi
+
+    cicd_ArtifactCommitAssociation=$(concat "$artifact_key" "$commit_key")
+    make_mutation cicd_artifact_commit_association "$cicd_ArtifactCommitAssociation"
+
+    cicd_Repository=$(jq \
+            '{data_artifact_repository,data_artifact_organization,data_artifact_source}' <<< $artifact_key
+            )
+    make_mutation cicd_repository "$cicd_Repository"
+
+    cicd_Organization=$(jq \
+            '{data_artifact_organization,data_artifact_source}' <<< $artifact_key
+            )
+    make_mutation cicd_organization "$cicd_Organization"
+}
+
+function make_mutation() {
+    log Calling Hasura rest endpoint $1 with payload $2
+
+    if !(($dry_run)); then
```

Comment:
perhaps log that data was not actually sent to Faros in the case of a dry_run.

Reply:
Done